src/mapping/nav2_map_server/test/integration/test_occ_grid_node.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <rclcpp/rclcpp.hpp>
+#include <memory>
+
+#include "nav2_map_server/map_reps/occupancy_grid.hpp"
+#include "test_constants/test_constants.h"
+
+// TODO(bpwilcox): test publisher
+
+class RclCppFixture
+{
+public:
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+RclCppFixture g_rclcppfixture;
+
+class TestNode : public ::testing::Test
+{
+public:
+  TestNode()
+  {
+    n_ = rclcpp::Node::make_shared("map_client_test");
+  }
+
+  template<class T>
+  typename T::Response::SharedPtr send_request(
+
+    rclcpp::Node::SharedPtr node,
+    typename rclcpp::Client<T>::SharedPtr client,
+    typename T::Request::SharedPtr request)
+  {
+    auto result = client->async_send_request(request);
+    // Wait for the result.
+    if (rclcpp::spin_until_future_complete(node, result) ==
+      rclcpp::executor::FutureReturnCode::SUCCESS)
+    {
+      return result.get();
+    } else {
+      return NULL;
+    }
+  }
+
+protected:
+  rclcpp::Node::SharedPtr n_;
+};
+
+TEST_F(TestNode, ResultReturned)
+{
+  auto req = std::make_shared<nav_msgs::srv::GetMap::Request>();
+  auto client = n_->create_client<nav_msgs::srv::GetMap>("occ_grid");
+  ASSERT_TRUE(client->wait_for_service(std::chrono::seconds(5)));
```

Comment:
Unless you feel that it is a failure if the service comes up in 6 seconds, then I'd just leave out a timeout here and wait forever.